contracts/protocol/libraries/logic/CalldataLogic.sol

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.10;
+
+import {DataTypes} from '../types/DataTypes.sol';
+
+/**
+ * @title CalldataLogic library
+ * @author Aave
+ * @notice Library to decode calldata, used to optimize calldata size in L2Pool for transaction cost reduction
+ */
+library CalldataLogic {
+  function decodeSupplyParams(mapping(uint256 => address) storage reservesList, bytes32 args)
+    internal
+    view
+    returns (
+      address,
+      uint256,
+      uint16
+    )
+  {
+    uint16 assetId;
+    uint256 amount;
+    uint16 referralCode;
+
+    assembly {
+      assetId := and(args, 0xFFFF)
+      amount := and(shr(16, args), 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
+      referralCode := and(shr(144, args), 0xFFFF)
+    }
+    return (reservesList[assetId], amount, referralCode);
+  }
+
+  function decodeSupplyWithPermitParams(
+    mapping(uint256 => address) storage reservesList,
+    bytes32 args
+  )
+    internal
+    view
+    returns (
+      address,
+      uint256,
+      uint16,
+      uint32,
+      uint8
+    )
+  {
+    uint32 deadline;
+    uint8 permitV;
+
+    bytes32 supplyParams;
+    assembly {
+      supplyParams := and(args, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
```

Comment:
i think the goal here was to use `supplyParams` as input parameter for `decodeSupplyParams` but `args` was used instead. I would remove the `supplyParams` as its not needed, `args` can be used directly